reana-ui/src/reducers.js

```diff
@@ -65,6 +72,18 @@ const detailsInitialState = {
   loadingDetails: false
 };
 
+const error = (state = errorInitialState, action) => {
```

Comment:
I have just one doubt now, with this change we will have now error data split into `auth.error` and `error`. Also, we have to call the reducers twice, to handle the error in the original part of the state tree (e.g. `workflow`) and in the `error` part:
```javascript
...
      dispatch(getError(err, USER_INFO_URL));
      dispatch({ type: WORKFLOWS_ERROR });
...
```
I wonder if we could somehow centralize this behavior (I've found [this](https://alexandrempsantos.com/sane-error-handling-react-redux/#after---how-to-solve-these-problems) which might be useful). What do you think?

Reply:
Very good remarks, thanks. With the `error` reducer I wanted to introduce a single place to store errors for the whole app. As a benefit it is very easy to know if there is something wrong with the app and to show it in the notification bar to the users. I really like the idea described in the link to unify the error handling, but as I understand it, it does not provide a central place to store errors. Each reducer will have it's own error object (because of combined reducers). Maybe there is a way to achieve it with some kind of similar approach?

Reply:
I see, thanks for the elaborated response! Maybe what we have is a naming issue then: if for instance `WORKFLOWS_ERROR` would be named `WORKFLOWS_FETCH_FAILED` it would make more sense in my eyes, in the end, what it does is just stopping the loader. This way the previous snipped would read:
```diff
...
      dispatch(getError(err, USER_INFO_URL));
-     dispatch({ type: WORKFLOWS_ERROR });
+     dispatch({ type: WORKFLOWS_FETCH_FAILED });
...
```

The problem is less evident with for instance [`USER_FETCH_ERROR`](https://github.com/reanahub/reana-ui/blob/97e0f77dcf44f0213c2b64ffdce2f349669bc9de/reana-ui/src/reducers.js#L108-L113) (or similar) since it handles the message too. Maybe the best would be to create an issue documenting this centralized approach you have started, so we can discuss it with @mvidalgarcia and standardize error management and notifications.

Reply:
Your suggestion makes mores sense to me as well, changed it accordingly. Also, wrote a comment in https://github.com/reanahub/reana-ui/issues/99 to come back to this discussion in the future.